pkg/docker/config/config.go

```diff
@@ -629,24 +698,28 @@ func decodeDockerAuth(conf dockerAuthConfig) (types.DockerAuthConfig, error) {
 	}, nil
 }
 
-// convertToHostname converts a registry url which has http|https prepended
-// to just an hostname.
-// Copied from github.com/docker/docker/registry/auth.go
-func convertToHostname(url string) string {
+// trimURL removes the prepended http|https from an URL as well as the registry
+// suffixes v1|v2.
+// Originally copied from github.com/docker/docker/registry/auth.go
+func trimURL(url string) string {
 	stripped := url
 	if strings.HasPrefix(url, "http://") {
 		stripped = strings.TrimPrefix(url, "http://")
 	} else if strings.HasPrefix(url, "https://") {
 		stripped = strings.TrimPrefix(url, "https://")
 	}
 
-	nameParts := strings.SplitN(stripped, "/", 2)
+	if strings.HasSuffix(stripped, "/v1") {
+		stripped = strings.TrimSuffix(stripped, "/v1")
+	} else if strings.HasSuffix(stripped, "/v2") {
+		stripped = strings.TrimSuffix(stripped, "/v2")
+	}
 
-	return nameParts[0]
+	return stripped
```

Comment:
Changing what we do with the old URLs:
- Doesn’t work anyway, this mangles user-intended `quay.io/v2ofmyproject` namespaces
- Seems unnecessarily risky for these extremely-legacy formats (remember from elsewhere that `docker login https://arbitray/url/with/unexpected/contents` was accepted for years, and worked)

I’m thinking (not sure) we don’t need to invoke `normalizeRegistry` for the new repo entries at all, and even if we did want to convert the hostname, splitting `normalizeRegistry` into two, one dealing with full `.dockercfg` URLs and one dealing with only hostnames, would be much safer.

Reply:
Okay, I splitted the http[s] trim part into a new function and added it to the path check in findAuthentication. This should not affect the original code path.